### ISSUE TITLE
Can't get past login page

### ISSUE THREAD

**Issue by Isaiah3133** [zeldridge/best-buy-sniper#11]
I've got the bot to add item to cart, but can't get account login info to save to browser for checkout.

**Comment by sgluss** [comment]
It looks like the bot is meant to checkout as guest: https://github.com/zeldridge/best-buy-sniper/blob/main/src/pages/bestbuy.ts#L68

**Comment by Isaiah3133** [comment]
> It looks like the bot is meant to checkout as guest: https://github.com/zeldridge/best-buy-sniper/blob/main/src/pages/bestbuy.ts#L68

Yeah, figured that after looking at gif. Had a few syntax problems that I fixed. Still can't get past guest checkout though.

**Comment by sgluss** [comment]
is it this issue? https://github.com/zeldridge/best-buy-sniper/issues/7

I wonder if bestbuy changed the site in the last few months, I see that this project hasn't been updated...

**Comment by Isaiah3133** [comment]
> is it this issue? #7
> 
> I wonder if bestbuy changed the site in the last few months, I see that this project hasn't been updated...

They have I heard. Aw man hopefully this is fixable.

**Comment by sgluss** [comment]
I'll take a look tonight, if they did change the site then the bot won't check out at all until this issue is fixed

**Comment by Isaiah3133** [comment]
> I'll take a look tonight, if they did change the site then the bot won't check out at all until this issue is fixed

Thanks man! I have little coding experience, trying what I can.

**Comment by sgluss** [comment]
looks like VScode supports debugging Node projects: https://code.visualstudio.com/docs/typescript/typescript-debugging

My guess is it's going to be a pretty simple adjustment to the playwright script.

I'd start by setting `config/prod/tasks.json` to try to purchase an in stock card, such as https://www.bestbuy.com/site/pny-nvidia-geforce-gt-710-verto-2gb-ddr3-pci-express-2-0-graphics-card-black/5092306.p?skuId=5092306

and of course, edit your CC info so it can't actually buy it! This should allow you to repro the issue for consistent testing, if you haven't already done so :)

**Comment by sgluss** [comment]
I was able to reproduce this issue consistently with the above steps, here are my logs:


│                                                     ││ main > 2021-01-21T18:30:59.214Z info: Navigating to cart                                                                     │
│                                                     ││ main > 2021-01-21T18:31:01.982Z info: Trying to checkout...                                                                  │
│                                                     ││ main > 2021-01-21T18:31:12.387Z warn: page.waitForSelector: Timeout 10000ms exceeded.                                        │
│                                                     ││ main > =========================== logs ===========================                                                          │
│                                                     ││ main > waiting for selector ".cia-guest-content .js-cia-guest-button" to be visible                                          │
│                                                     ││ main > ============================================================

looks like the guest checkout selector needs to be updated

**Comment by sgluss** [comment]
OK fixed the selectors, with the test procedure above my system is now able to complete the purchase flow for this task:
```
        "products": [
          {
            "searchText": "NVIDIA GeForce GT 710 VERTO 2GB",
            "sku": "5092306",
            "model": "VCGGT7102XPB-BB",
            "productName": "PNY - NVIDIA GeForce GT 710 VERTO 2GB DDR3 PCI Express 2.0 Graphics Card - Black",
            "productPage": "/site/pny-nvidia-geforce-gt-710-verto-2gb-ddr3-pci-express-2-0-graphics-card-black/5092306.p?skuId=5092306"
          }
        ]
```
Here is the fix:
https://github.com/zeldridge/best-buy-sniper/pull/13

@Isaiah3133 can you check if this PR fixes the issue for you?

**Comment by Isaiah3133** [comment]
What about when the Item isn't instock, will it still leave the item in the cart? then add another, and so on?

**Comment by Isaiah3133** [comment]
thanks

**Comment by sgluss** [comment]
> What about when the Item isn't instock, will it still leave the item in the cart? then add another, and so on?

I noticed the bot would add multiple items to the cart, but only if the item is in stock, AND the checkout attempt fails. This should not be a problem now that the checkout flow is fixed.

If the item is not in stock, the bot will not be able to add it to the cart, and will simply retry until it can.

**Comment by sgluss** [comment]
I did notice that bestbuy will correctly recognize the checkout as bot driven, I think that will be the main issue once we get this fix in (I also have no idea how we'd fix that, though occasionally bestbuy doesn't notice that we're botting 😅 )
